chips/msp432/src/uart.rs

```diff
@@ -77,15 +75,14 @@ pub struct Uart<'a> {
 }
 
 impl<'a> Uart<'a> {
-    pub(crate) const fn new(
-        regs: StaticRef<UsciARegisters>,
+    pub const fn new(
         tx_dma_chan: usize,
         rx_dma_chan: usize,
         tx_dma_src: u8,
         rx_dma_src: u8,
-    ) -> Uart<'static> {
-        Uart {
-            registers: regs,
+    ) -> Self {
+        Self {
+            registers: usci::USCI_A0_BASE,
             clock_frequency: DEFAULT_CLOCK_FREQ_HZ,
 
```

Comment:
This constructor seems wrong to me since the reference to the registers isn't passed as argument anymore. If other UARTs are instantiated they all point to the same registers.

Reply:
That's a good point, I didn't realize the MSP432 had multiple UARTs. I'll fix